virtualizarr/accessor.py

```diff
@@ -48,10 +52,17 @@ def to_icechunk(
 
         If `append_dim` is provided, the virtual dataset will be appended to the existing IcechunkStore along the `append_dim` dimension.
 
+        If `last_updated_at` is provided, it will be used as a checksum for any virtual chunks written to the store with this operation. At read time, if any of the virtual chunks have been updated since this provided datetime, an error will be raised.
+            This protects against reading outdated virtual chunks that have been updated since the last read.
+
         Parameters
         ----------
         store: IcechunkStore
         append_dim: str, optional
+            When provided, specifies the dimension along which to append the virtual dataset.
+        last_updated_at: datetime, optional
+            When provided, uses provided datetime as a checksum for any virtual chunks written to the store with this operation.
```

Comment:
This is a cool feature! But without an example I'm unclear if the intended usage pattern is to pass the current time or the time at which the referenced archival data was written. In fact is there a reason the default value isn't just to use the current time?

An example either in the docstring (could add an `Examples` section) or in the usage docs would help.

Reply:
Also what is the behaviour when the default of `None` is passed?

Reply:
when none is passed there is no check. so thats a better default for those who dont want or care about this check

Reply:
Okay - let's just document in the docstring that there is no check then.